### DISCUSSION TITLE
Screen Space Shadow Map

### DISCUSSION THREAD

**Discussion by errissa** [mitsuba-renderer/mitsuba3#551] [Q&A]
I would like to generate an image recording the visibility of the objects in a scene to the emitters in a scene, essentially a screen space shadow map, but I haven't been able to determine if that's even possible with Mitsuba or if perhaps some custom BSDF is necessary. Any ideas?

**Comment by njroussel** [comment]
Hi @errissa 

I have practically no knowledge about screen space shadow maps. If I understood it correctly from your description and a quick google search, you want a [`direct` integrator ](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#direct-illumination-integrator-direct) with the difference that you only want to measure the visibility rather than the radiance.

I assume you still care about how the BSDFs scatter light? If not you would need to replace every material in your scene.

I think the way to do this is a custom `Integrator`, essentially a copy of `direct` with a few changes to replace any radiance measures by visibility measures.

I would encourage you to take a look at [this tutorial](https://mitsuba.readthedocs.io/en/stable/src/rendering/scripting_renderer.html#Ambient-occlusion). It might actually be easier to script what you're trying to achieve in a tiny renderer/integrator rather than dealing with some of the `Integrator` interface complexities.
